tests/test_culling.py

```diff
@@ -304,7 +304,13 @@ def test_culled_host_is_removed(
     assert db_get_host(created_host.id)
 
     threadctx.request_id = UNKNOWN_REQUEST_ID_VALUE
-    host_reaper_run(inventory_config, mock.Mock(), db.session, event_producer_mock)
+    host_reaper_run(
+        inventory_config,
+        mock.Mock(),
+        db.session,
+        event_producer_mock,
+        shutdown_handler=mock.Mock(**{"shut_down.side_effect": (False, False)}),
```

Comment:
We can even go with _return_value_ here. We are not testing the shutdown handler, so no interruptions at all.

```suggestion
        shutdown_handler=mock.Mock(**{"shut_down.return_value": False}),
```